### PR TITLE
Move sun.net Resolver down

### PR DESCRIPTION
**Pull request by kingle**
Moving the sun.net.dns.ResolverConfiguration down the ordering

### CODE REVIEW

**Comment by kingle** [comment]
The driving force is that on Unix-like systems, the preference of order would be to use resolv.conf before trying the sun implementation, for ndots.  Your point is taken about Windows.  If this is changed to try the sun resolver on Windows first and fallback to ipconfig, would that suffice?

As far as the SPI work, I was hoping the OpenJDK itself would progress further on the (almost identical) interface they initially suggested: https://bugs.openjdk.java.net/browse/JDK-8211216

**Comment by ibauersachs** [comment]
`resolv.conf` for UNIX-like is fine with me. Forget about hoping the JDK will act on this anytime soon, and we need a solution for JDK 11 anyway since this is what Ubuntu 18.04 and Debian Buster are shipping in their stable/LTS releases.

File: src/main/java/org/xbill/DNS/ResolverConfig.java [hunk 1]
```diff
@@ -52,21 +52,25 @@
 ResolverConfig() {
 	if (findProperty())
 		return;
-	if (findSunJVM())
-		return;
+	boolean attemptedSun = false;
 	if (servers == null || searchlist == null) {
```
Comment (ibauersachs) [inline src/main/java/org/xbill/DNS/ResolverConfig.java]: This check is now useless - findProperty is ony true when either servers or searchList is null

File: src/main/java/org/xbill/DNS/ResolverConfig.java [hunk 2]
```diff
@@ -52,21 +52,25 @@
...
-			findNetware();
-		} else if (vendor.contains("Android")) {
-			findAndroid();
-		} else {
-			findUnix();
-		}
+			if (findSunJVM() || findWin())
+				return;
+			else
+				attemptedSun = true;
```
Comment (ibauersachs) [inline src/main/java/org/xbill/DNS/ResolverConfig.java]: I don't think `attemptedSun` makes sense like this. It's only used once, so the code would be simpler to read if you just add

    else
        findSunJVM();

as the general fallback if none of the OS checks match.

**Pull request closed without merging**